tests/unit/confluence/test_search.py

```diff
@@ -265,3 +267,319 @@ def test_search_general_exception(self, search_mixin):
+            (None, 0),  # None response case
+            ({"results": []}, 0),  # Empty results case
+        ],
+    )
+    def test_search_user_edge_cases(self, search_mixin, mock_response, expected_length):
+        """Test search_user handling of edge cases in API responses."""
+        search_mixin.confluence.get.return_value = mock_response
+
+        # Act
+        results = search_mixin.search_user('user.fullname ~ "Test"')
+
+        # Assert
+        assert isinstance(results, list)
+        assert len(results) == expected_length
+
+    # You can also parametrize the regular search method exception tests:
+    @pytest.mark.parametrize(
+        "exception_type,exception_args,expected_result",
+        [
+            (requests.RequestException, ("API error",), []),
+            (ValueError, ("Value error",), []),
+            (TypeError, ("Type error",), []),
+            (Exception, ("General error",), []),
+            (KeyError, ("Missing key",), []),
+        ],
+    )
+    def test_search_exception_handling(
+        self, search_mixin, exception_type, exception_args, expected_result
+    ):
+        """Test search handling of various exceptions that return empty list."""
+        # Mock the exception
+        search_mixin.confluence.cql.side_effect = exception_type(*exception_args)
+
+        # Act
+        results = search_mixin.search("error query")
+
+        # Assert
+        assert isinstance(results, list)
+        assert results == expected_result
+
+    # Parametrize CQL query tests:
+    @pytest.mark.parametrize(
+        "query,limit,expected_params",
+        [
+            (
+                'user.fullname ~ "Test"',
+                10,
+                {"cql": 'user.fullname ~ "Test"', "limit": 10},
+            ),
+            (
+                'user.email ~ "test@example.com"',
+                5,
+                {"cql": 'user.email ~ "test@example.com"', "limit": 5},
+            ),
+            (
+                'user.fullname ~ "John" AND user.email ~ "@company.com"',
+                15,
+                {
+                    "cql": 'user.fullname ~ "John" AND user.email ~ "@company.com"',
+                    "limit": 15,
+                },
+            ),
+        ],
+    )
+    def test_search_user_api_parameters(
+        self, search_mixin, query, limit, expected_params
+    ):
+        """Test that search_user calls the API with correct parameters."""
+        # Mock successful response
+        search_mixin.confluence.get.return_value = {
+            "results": [],
+            "start": 0,
+            "limit": limit,
+            "totalSize": 0,
+        }
+
+        # Act
+        search_mixin.search_user(query, limit=limit)
+
+        # Assert API was called with correct parameters
+        search_mixin.confluence.get.assert_called_once_with(
+            "rest/api/search/user", params=expected_params
+        )
+
+    def test_search_user_api_call_parameters(self, search_mixin):
```

Comment:
It seems that `test_search_user_api_call_parameters` is now redundant, as its case is covered by the above test. consider removing this